core/modules/views/includes/cache.inc

```diff
@@ -127,8 +127,27 @@ function _views_data_process_entity_types(&$data) {
 function _views_fetch_plugin_data($type = NULL, $plugin = NULL, $reset = FALSE) {
   static $cache = NULL;
   if (!isset($cache) || $reset) {
+    // Load necessary code once.
     views_include_handlers();
-    $cache = views_discover_plugins();
+
+    // Because plugin data contains translated strings, and as such can be
+    // expensive to build, the results are cached per language.
+    global $language;
+    $cache_key = 'views_plugin_data:' . $language->langcode;
+    if ($reset) {
+      $cache = NULL;
+    }
```

Comment:
I changed this line from the port. Looking at what was committed to Drupal, I can't see how it would work properly because the static cache value is still used even if the `$reset` parameter is passed in. Perhaps our test coverage was simply different than Drupal's and caught this problem which may have existed in the original patch.